src/values/instruction_value.rs

```diff
@@ -665,6 +665,8 @@ impl<'ctx> InstructionValue<'ctx> {
     /// Determines whether or not this `Instruction` has any associated metadata
     /// `kind_id`.
     pub fn set_metadata(self, metadata: MetadataValue<'ctx>, kind_id: u32) {
+        assert!(metadata.is_node());
```

Comment:
You mentioned `add_global_metadata` can't be provided a string; but why was this added here? Does a string also trigger an LLVM assertion here?

Reply:
And if so, this should become a result

Reply:
Rather than being a result, can we make it an `Option`? It will make things a bit easier.

If we prefer a result, what the Error type should be?

Reply:
Done